polyply/src/load_library.py

```diff
@@ -13,75 +13,112 @@
+    elif not is_lib_file:
+        msg = "Cannot parse user provided file with extension {file_extension}."
+        raise IOError(msg)
 
-def _resolve_lib_paths(lib_names, data_path):
+
+def _resolve_lib_files(lib_names, data_path):
     """
-    select the appropiate files from data_path
+    select the appropiate files from a file path
     according to library names given.
 
     Parameters
     ----------
-    lib_names: list[`os.path`]
+    lib_names: list[`pathlib.Path`]
         list of library names
     data_path:
         location of library files
     """
     files = []
-    for name in lib_names:
-        directory = os.path.join(data_path, name)
-        for _file in os.listdir(directory):
-            _file = os.path.join(directory, _file)
-            files.append(_file)
+    data_path = Path(data_path)
+    for name in lib_names or []:
+        directory = data_path.joinpath(name)
+        for file_ in os.listdir(directory):
+            file_path = directory.joinpath(file_)
+            files.append(file_path)
     return files
 
 
-def read_ff_from_files(paths, force_field):
+def read_options_from_files(paths, storage_object, file_parsers):
     """
-    read the input files for the defintion of blocks and links.
+    read the input files for the definition of blocks and links.
 
     Parameters
     ----------
-    paths: list
-           List of vaild file paths
-    force_field: class:`vermouth.force_field.ForceField`
-
-    Returns
-    -------
-    force_field: class:`vermouth.force_field.ForceField`
-       updated forcefield
+    paths: list[`pathlib.Path`]
+           List of provided file paths
+    storage_object: polyply.src.topology.Topology or vermouth.ForceField
+    file_parsers: dict
+        dictionary of available file parsers
 
     """
 
-    def wrapper(parser, path, force_field):
+    def parse_file(parser, path, storage_object):
         with open(path, 'r') as file_:
             lines = file_.readlines()
-            parser(lines, force_field=force_field)
+            parser(lines, storage_object)
 
-    for path in paths:
-        path = pathlib.Path(path)
-        file_extension = path.suffix.casefold()[1:]
+    lib_files, user_files = paths
+    for path in user_files + lib_files or []:
+        is_lib_file = path in lib_files
+        parser = get_parser(path, file_parsers, is_lib_file)
+        if parser:
+            parse_file(parser, path, storage_object)
 
-        try:
-            parser = FORCE_FIELD_PARSERS[file_extension]
-            wrapper(parser, path, force_field)
-        except KeyError:
-            raise IOError(
-                "Cannot parse file with extension {}.".format(file_extension))
 
-    return force_field
+def load_build_files(topology, lib_names, build_file):
+    """
+    Load build file options and molecule templates into topology.
+
+    Parameters
+    ----------
+    topology: :class:`polyply.src.topology`
+    lib_names: list[str]
```

Comment:
should this be able to handle mutliple ffs. What if they contain conflicting information?